libs/partners/cohere/langchain_cohere/chat_models.py

```diff
@@ -60,33 +61,32 @@ def get_cohere_chat_request(
     Returns:
         The request for the Cohere chat API.
     """
-    documents = (
-        None
-        if "source_documents" not in kwargs
-        else [
-            {
-                "snippet": doc.page_content,
-                "id": doc.metadata.get("id") or f"doc-{str(i)}",
-            }
-            for i, doc in enumerate(kwargs["source_documents"])
-        ]
-    )
-    kwargs.pop("source_documents", None)
-    maybe_connectors = connectors if documents is None else None
+    additional_kwargs = messages[-1].additional_kwargs
+
+    # cohere SDK will fail loudly if both connectors and documents are provided
+    formatted_docs = [
+        {
+            "text": doc.page_content,
+            "id": doc.metadata.get("id") or f"doc-{str(i)}",
+        }
+        for i, doc in enumerate(additional_kwargs.get("documents", []))
+    ] or documents
```

Comment:
should we throw an error if both `documents` and `additional_kwargs["documents"]` is set?

Reply:
Added. Good suggestion.